### DISCUSSION TITLE
seq cannot unlock any user when Active Directory is enabled

### DISCUSSION THREAD

**Discussion by zengli250** [datalust/seq-tickets#1838] [Help]
Hello - We have a legacy Seq system whose admin left the company. Currently seq auth shows a few users, and we'd like to unlock password for these users (and preferably grant admin privileges to these users). However, this fails due to Active Directory being enabled:

> seq unlock -u admin -p newpassword
Could not reset the password: Active directory authentication is enabled; passwords must be reset via the directory.

However, we need admin permissions in order to access and disable the Active Directory, so this is a circular situation.

Is there any way to restore admin access to such a system? Thank you,
-Lingkai

**Comment by liammclennan** [comment]
Hi @zengli250 

Can you please tell me which version of Seq you have?

**Comment by zengli250** [comment]
Hi Liam, the version is 5.1.3200. Thanks,
-Lingkai

From: Liam McLennan ***@***.***>
Sent: Thursday, May 11, 2023 6:53 PM
To: datalust/seq-tickets ***@***.***>
Cc: Zeng, Lingkai ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [datalust/seq-tickets] seq cannot unlock any user when Active Directory is enabled (Discussion #1838)


Hi @zengli250<https://urldefense.com/v3/__https:/github.com/zengli250__;!!OD3VmZw!Z_nU1FlPupceXxahq2a9m-wurIPXa-Jbdf8NVpxJMee5gQ_fy5_a9id0uxiK_an-8eHob-hhFbncConeerDglgPThQ$>

Can you please tell me which version of Seq you have?

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/datalust/seq-tickets/discussions/1838*discussioncomment-5878627__;Iw!!OD3VmZw!Z_nU1FlPupceXxahq2a9m-wurIPXa-Jbdf8NVpxJMee5gQ_fy5_a9id0uxiK_an-8eHob-hhFbncConeerCyhBJTzg$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/A7YRTLI6SAVE2HWPHYPMPILXFVUU7ANCNFSM6AAAAAAX6HRKW4__;!!OD3VmZw!Z_nU1FlPupceXxahq2a9m-wurIPXa-Jbdf8NVpxJMee5gQ_fy5_a9id0uxiK_an-8eHob-hhFbncConeerCvawidkw$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>


*********************

IMPORTANT NOTICE: This e-mail message and any attachments are intended solely for the use of the individual or entity to which it is addressed and may contain information that is privileged, confidential, proprietary or otherwise protected from disclosure. If you are not the intended recipient, you are hereby notified that any dissemination, distribution, copying or other use of this message or its attachments is strictly prohibited. If you have received this message in error, please notify the sender immediately and permanently delete this message and any attachments. Nothing contained in this message and/or any attachment(s) constitutes investment advice or a recommendation to purchase or sell any financial product and you may not rely upon the sender or Soros Fund Management LLC or its affiliates for any investment advice. We may monitor and retain email communications for legal and regulatory compliance purposes.

**Reply by nblumhardt** [reply]
Hi @zengli250! Thanks for the info.

Seq 5.1 (while it was a great release very close to my heart :-)) is now very, very far out of support and doesn't receive any bug fixes or security patches. Upgrading to Seq 2023.1 is worthwhile, as soon as you're able to. Instructions can be found in https://docs.datalust.co/docs/upgrading.

On 5.1, the command to reset authentication is:

```
seq auth --basic -u admin -p newpassword
```

Let me know if this helps,
Nick

**Comment by zengli250** [comment]
Hi Nicholas, I will try upgrading, though currently there seems to be Page Not Found error when I go to the upgrade link.

Also, I tried the seq auth command which did not work due to Active Directory being enabled. Is there a way to disable Active Directory without admin permissions? As is, it’s a circular loop where I need admin permissions to disable Active Directory, but because Active Directory I cannot run the seq auth to gain admin permissions. Is there a mechanism to disable Active Directory without having admin permission? The admin who originally set this up is no longer reachable. Thanks,

-Lingkai

From: Nicholas Blumhardt ***@***.***>
Sent: Thursday, May 11, 2023 10:00 PM
To: datalust/seq-tickets ***@***.***>
Cc: Zeng, Lingkai ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [datalust/seq-tickets] seq cannot unlock any user when Active Directory is enabled (Discussion #1838)


Hi @zengli250<https://urldefense.com/v3/__https:/github.com/zengli250__;!!OD3VmZw!fQF0qfIiu5zZyrAyutlrCfKcvRcdAATOGcngbcpKGxTDWicNTr8JgCTFfqF7wEOSREfWv3rDhccqwr6X-6VEgv0UyQ$>! Thanks for the info.

Seq 5.1 (while it was a great release very close to my heart :-)) is now very, very far out of support and doesn't receive any bug fixes or security patches. Upgrading to Seq 2023.1 is worthwhile, as soon as you're able to. Instructions can be found in https://docs.datalust.co/doc/upgrading<https://urldefense.com/v3/__https:/docs.datalust.co/doc/upgrading__;!!OD3VmZw!fQF0qfIiu5zZyrAyutlrCfKcvRcdAATOGcngbcpKGxTDWicNTr8JgCTFfqF7wEOSREfWv3rDhccqwr6X-6X5ChTZnA$>.

On 5.1, the command to reset authentication is:

seq auth --basic -u admin -p newpassword

Let me know if this helps,
Nick

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/datalust/seq-tickets/discussions/1838*discussioncomment-5879750__;Iw!!OD3VmZw!fQF0qfIiu5zZyrAyutlrCfKcvRcdAATOGcngbcpKGxTDWicNTr8JgCTFfqF7wEOSREfWv3rDhccqwr6X-6VKwYDTJA$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/A7YRTLNRB3VAHBCGEUI7QB3XFWKSHANCNFSM6AAAAAAX6HRKW4__;!!OD3VmZw!fQF0qfIiu5zZyrAyutlrCfKcvRcdAATOGcngbcpKGxTDWicNTr8JgCTFfqF7wEOSREfWv3rDhccqwr6X-6UXVz4Z8A$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>


*********************

IMPORTANT NOTICE: This e-mail message and any attachments are intended solely for the use of the individual or entity to which it is addressed and may contain information that is privileged, confidential, proprietary or otherwise protected from disclosure. If you are not the intended recipient, you are hereby notified that any dissemination, distribution, copying or other use of this message or its attachments is strictly prohibited. If you have received this message in error, please notify the sender immediately and permanently delete this message and any attachments. Nothing contained in this message and/or any attachment(s) constitutes investment advice or a recommendation to purchase or sell any financial product and you may not rely upon the sender or Soros Fund Management LLC or its affiliates for any investment advice. We may monitor and retain email communications for legal and regulatory compliance purposes.

**Reply by nblumhardt** [reply]
Thanks for your reply, sorry about the broken link - I've edited my answer above to fix it.

I'm not sure what will have gone wrong running the "auth" command. [In Seq 5.1, that command](https://docs.datalust.co/v5.1/docs/server-command-line#auth) exists solely for the purpose of changing the authentication provider, it shouldn't have any problem disabling Active Directory authentication.

Did you possibly type "unlock" instead of "auth" by mistake? If not, can you please send a capture of the output the command produced?

**Comment by zengli250** [comment]
Thanks Nicholas! I tried this on one of the boxes and it seemed to work, you’re right I probably tried unlock instead of auth earlier. Will let you know in case I see any other issues. Best,
-Lingkai

From: Nicholas Blumhardt ***@***.***>
Sent: Friday, May 12, 2023 5:57 PM
To: datalust/seq-tickets ***@***.***>
Cc: Zeng, Lingkai ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [datalust/seq-tickets] seq cannot unlock any user when Active Directory is enabled (Discussion #1838)


Thanks for your reply, sorry about the broken link - I've edited my answer above to fix it.

I'm not sure what will have gone wrong running the "auth" command. In Seq 5.1, that command<https://urldefense.com/v3/__https:/docs.datalust.co/v5.1/docs/server-command-line*auth__;Iw!!OD3VmZw!ajLK0DB5AsFK0lCXpi1qhhGRU5UbBY__AuXfG9YaxEqfciLHBPan6qFUcQNs39hJHzyDiGZvIWqKtnsvbjIjfgC0vA$> exists solely for the purpose of changing the authentication provider, it shouldn't have any problem disabling Active Directory authentication.

Did you possibly type "unlock" instead of "auth" by mistake? If not, can you please send a capture of the output the command produced?

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/datalust/seq-tickets/discussions/1838*discussioncomment-5889030__;Iw!!OD3VmZw!ajLK0DB5AsFK0lCXpi1qhhGRU5UbBY__AuXfG9YaxEqfciLHBPan6qFUcQNs39hJHzyDiGZvIWqKtnsvbjKbxz_Z1Q$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/A7YRTLOOT4TXOYEOICWJRDLXF2WZ3ANCNFSM6AAAAAAX6HRKW4__;!!OD3VmZw!ajLK0DB5AsFK0lCXpi1qhhGRU5UbBY__AuXfG9YaxEqfciLHBPan6qFUcQNs39hJHzyDiGZvIWqKtnsvbjInNBwAYg$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>


*********************

IMPORTANT NOTICE: This e-mail message and any attachments are intended solely for the use of the individual or entity to which it is addressed and may contain information that is privileged, confidential, proprietary or otherwise protected from disclosure. If you are not the intended recipient, you are hereby notified that any dissemination, distribution, copying or other use of this message or its attachments is strictly prohibited. If you have received this message in error, please notify the sender immediately and permanently delete this message and any attachments. Nothing contained in this message and/or any attachment(s) constitutes investment advice or a recommendation to purchase or sell any financial product and you may not rely upon the sender or Soros Fund Management LLC or its affiliates for any investment advice. We may monitor and retain email communications for legal and regulatory compliance purposes.
